src/Arcus.Security.Core/ISecretProvider.cs

```diff
@@ -200,7 +205,7 @@ public static SecretResult Interrupted(string failureMessage, Exception failureC
         /// <summary>
         /// Gets the secret value that was retrieved from the secret provider.
         /// </summary>
-        public string Name => IsSuccess ? _secretName : throw new InvalidOperationException($"[Arcus] cannot get secret name as the secret retrieval failed: {_failureMessage}", _failureCause);
+        public string Name { get; }
```

Comment:
There's a bit of a discrepancy imho between this property-name (`Name`) and the parameter-names of the factory-methods (`secretName`).  Should this property also be named `SecretName`?

Reply:
Indeed, I agree that it would add noise.
Renaming the parameters would be a better option then; it's not a must for me, I'll let you decide :)